modules/plugin-search.php

```diff
@@ -0,0 +1,486 @@
+			}
+		}
+		return $result;
+	}
+
+	/**
+	 * Take a raw search query and return something a bit more standardized and
+	 * easy to work with.
+	 *
+	 * @param  String $term The raw search term
+	 * @return String A simplified/sanitized version.
+	 */
+	private function sanitize_search_term( $term ) {
+		$term = strtolower( urldecode( $term ) );
+
+		// remove non-alpha/space chars.
+		$term = preg_replace( '/[^a-z ]/', '', $term );
+
+		// remove strings that don't help matches.
+		$term = trim( str_replace( array( 'jetpack', 'jp', 'free', 'wordpress' ), '', $term ) );
+
+		return $term;
+	}
+
+	/**
+	 * Callback function to sort the array of modules by the sort option.
+	 */
+	private function by_sorting_option( $m1, $m2 ) {
+		return $m1['sort'] - $m2['sort'];
+	}
+
+	/**
+	 * Builds a URL to purchase and upgrade inserting the site fragment and the affiliate code if it exists.
+	 *
+	 * @param string $feature Module slug (or forged one for extra features).
+	 *
+	 * @since 7.1.0
+	 *
+	 * @return string URL to upgrade.
+	 */
+	private function get_upgrade_url( $feature ) {
+		$site_raw_url = Jetpack::build_raw_urls( get_home_url() );
+		$affiliateCode = Jetpack_Affiliate::init()->get_affiliate_code();
+		$user = wp_get_current_user()->ID;
+		return "https://jetpack.com/redirect/?source=plugin-hint-upgrade-$feature&site=$site_raw_url&u=$user" .
+		       ( $affiliateCode ? "&aff=$affiliateCode" : '' );
+	}
+
+	/**
+	 * Modify the URL to the feature settings, for example Publicize.
+	 * Sharing is included here because while we still have a page in WP Admin,
+	 * we prefer to send users to Calypso.
+	 *
+	 * @param string $feature
+	 *
+	 * @since 7.1.0
+	 *
+	 * @return string
+	 */
+	private function get_configure_url( $feature ) {
+		$url = Jetpack::module_configuration_url( $feature );
+		$siteFragment = Jetpack::build_raw_urls( get_home_url() );
+		switch ( $feature ) {
+			case 'sharing':
+			case 'publicize':
+				$url = "https://wordpress.com/sharing/$siteFragment";
+				break;
+			case 'seo-tools':
+				$url = "https://wordpress.com/settings/traffic/$siteFragment#seo";
+				break;
+			case 'google-analytics':
+				$url = "https://wordpress.com/settings/traffic/$siteFragment#analytics";
+				break;
+			case 'wordads':
+				$url = "https://wordpress.com/ads/settings/$siteFragment";
+				break;
+		}
+		return esc_url( $url );
+	}
+
+	/**
+	 * Put some more appropriate links on our custom result cards.
+	 */
+	public function insert_module_related_links( $links, $plugin ) {
+		if ( self::$slug !== $plugin['slug'] ) {
+			return $links;
+		}
+
+		// By the time this filter is applied, self_admin_url was already applied and we don't need it anymore.
+		remove_filter( 'self_admin_url', array( $this, 'plugin_details' ) );
+
+		$links = array();
+
+		// Jetpack installed, active, feature not enabled; prompt to enable.
+		if (
+			(
+				Jetpack::is_active() ||
+				(
+					Jetpack::is_development_mode() &&
```

Comment:
Do we need this here, given that we don't seem to initiate anything when the plugin is not connected to WordPress.com?
https://github.com/Automattic/jetpack/pull/10611/files#diff-f33103bd412ed49a26cdf8f3756f0e58R11